### ISSUE TITLE
Bug: nn.Linear bias and weight gradients are None after backward()

### ISSUE THREAD

**Issue by meigel** [geohot/tinygrad#15242]
## Bug Description

When using `nn.Linear` from tinygrad, neither `weight` nor `bias` receive gradients after calling `.backward()`.

## Minimal Reproduction

```python
import sys
sys.path.insert(0, 'tinygrad')
from tinygrad import Tensor
import tinygrad.nn as nn

# Create linear layer with bias
fc = nn.Linear(4, 3, bias=True)

x = Tensor.randn(5, 4, requires_grad=True)

# Forward pass
y = fc(x)

# Backward pass  
loss = y.sum()
loss.backward()

print(f'weight.grad exists: {fc.weight.grad is not None}')  # False - BUG!
print(f'bias.grad exists: {fc.bias.grad is not None}')      # False - BUG!
```

## Expected Behavior

Both `weight.grad` and `bias.grad` should be `Tensor` objects after backprop.

## Actual Behavior

Both are `None`.

## Additional Notes

- This affects MLP implementations that rely on `nn.Linear` layers
- Manual tensor operations work correctly: `(x @ W.T + b).sum().backward()` produces correct gradients
- The issue appears to be in how `nn.Linear` implements its forward pass with the bias term

## Environment

- tinygrad (latest from submodule)
- Python 3.12
- Linux

**Comment by chenyuxyz** [comment]
```
  Root Cause: nn.Linear creates its weight and bias via Tensor.uniform() which passes requires_grad=None to requires_grad_(). This leaves
  requires_grad as None (falsy).

  The backward() method (line 1061-1062) filters tensors by t.requires_grad — since None is falsy, weight/bias are excluded from gradient
  computation.

  The intended workflow in tinygrad is that requires_grad gets set to True when parameters are passed to an optimizer (see optim.py line
  12-14). But if you call .backward() directly without an optimizer (as the issue reporter does), the gradients are never computed for these
   parameters.

  This is a design difference from PyTorch, where nn.Parameter always has requires_grad=True. In tinygrad, the user must either:
  1. Pass parameters to an optimizer first (which sets requires_grad=True), or
  2. Manually set requires_grad=True on each parameter
```

**Comment by meigel** [comment]
I see, thanks for the clarification.
